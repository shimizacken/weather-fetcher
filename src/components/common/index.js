export { Button } from './button/Button';
export { SearchBox } from './searchBox/SearchBox';
export { RadioButton } from './radioButton/RadioButton';
export { Loader } from './loader/Loader';
export { Row } from './table';
export { Modal, ConfirmationModal } from './modal';
