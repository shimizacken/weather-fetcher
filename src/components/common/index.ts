export {Button} from './button/Button';
export {ButtonType} from './button/buttonType';
export {SearchBox} from './searchBox/SearchBox';
export {RadioButton} from './radioButton/RadioButton';
export {Loader} from './loader/Loader';
export {Table, Row, Column} from './table';
export {Modal, ConfirmationModal} from './modal';
export {BackButton} from './button';
export {TextInput} from './textInput/TextInput';
export {Logo} from './logo/Logo';
