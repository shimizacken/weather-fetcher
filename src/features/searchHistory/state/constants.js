export const SET_SEARCH_HISTORY = 'search-history/set';
export const START_ADD_TO_SEARCH_HISTORY = 'search-history/add-item/start';
export const ADD_TO_SEARCH_HISTORY_END = 'search-history/add-item/end';
export const REMOVE_ITEM_FROM_HISTORY = 'search-history/remove-item';
export const END_REMOVE_ITEM_FROM_HISTORY = 'search-history/remove-item/end';
export const CLEAR_ALL_SEARCH_HISTORY = 'search-history/remove-all-items';
