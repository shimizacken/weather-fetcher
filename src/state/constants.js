import uuid from 'uuid';

export const RANDOM_ACTION_TYPE = `@@rand/action/${uuid.v4()}`;

export const pending = 'pending';
export const success = 'success';
export const error = 'error';

/**
 * Append '_pending' suffix to given action type.
 */
export const pendingType = (actionType) => `${actionType}_${pending}`;

/**
 * Append '_success' suffix to given action type.
 */
export const successType = (actionType) => `${actionType}_${success}`;

/**
 * Append '_error' suffix to given action type.
 */
export const errorType = (actionType) => `${actionType}_${error}`;
