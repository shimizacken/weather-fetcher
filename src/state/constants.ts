import uuid from 'uuid';

export const RANDOM_ACTION_TYPE: string = `@@rand/action/${uuid.v4()}`;

export const PENDING: string = 'pending';
export const SUCCESS: string = 'success';
export const ERROR: string = 'error';

/**
 * Append '_pending' suffix to given action type.
 */
export const pendingType = (actionType: string): string => `${actionType}_${PENDING}`;

/**
 * Append '_success' suffix to given action type.
 */
export const successType = (actionType: string): string => `${actionType}_${SUCCESS}`;

/**
 * Append '_error' suffix to given action type.
 */
export const errorType = (actionType: string): string => `${actionType}_${ERROR}`;
