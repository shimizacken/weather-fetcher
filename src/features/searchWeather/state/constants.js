import { pendingType, successType, errorType } from 'app/state/constants';

export const FETCH_WEATHER = 'weather/fetch';
export const SET_WEATHER = 'weather/set';

export const FETCH_WEATHER_PENDING = pendingType(FETCH_WEATHER);
export const FETCH_WEATHER_SUCCESS = successType(FETCH_WEATHER);
export const FETCH_WEATHER_ERROR = errorType(FETCH_WEATHER);
