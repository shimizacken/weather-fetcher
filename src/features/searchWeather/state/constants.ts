import { pendingType, successType, errorType } from 'app/state/constants';

export const FETCH_WEATHER: string = 'weather/fetch';
export const SET_WEATHER: string = 'weather/set';
export const SET_WEATHER_END: string = 'weather/set/end';

export const FETCH_WEATHER_PENDING: string = pendingType(FETCH_WEATHER);
export const FETCH_WEATHER_SUCCESS: string = successType(FETCH_WEATHER);
export const FETCH_WEATHER_ERROR: string = errorType(FETCH_WEATHER);

export const SET_FETCH_WEATHER_FLAG: string = 'weather/set-fetch-flag';
