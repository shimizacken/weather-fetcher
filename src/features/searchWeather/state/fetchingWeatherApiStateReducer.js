import { FETCH_WEATHER_PENDING, FETCH_WEATHER_SUCCESS, FETCH_WEATHER_ERROR } from './constants';

const initialState = {
  isFetching: false,
  isError: false,
  errorMessage: ''
};

export const fetchingWeatherApiState = (state = initialState, action) => {
  if (action.type === FETCH_WEATHER_SUCCESS) {
    return initialState;
  }

  if (action.type === FETCH_WEATHER_PENDING) {
    return {
      ...state,
      isFetching: true,
      isError: false,
      errorMessage: ''
    };
  }

  if (action.type === FETCH_WEATHER_ERROR) {
    return {
      ...state,
      isFetching: false,
      isError: true,
      errorMessage: action?.payload?.message
    };
  }

  return state;
};
