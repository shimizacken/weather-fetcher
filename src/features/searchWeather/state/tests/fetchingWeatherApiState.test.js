import { randomAction } from 'app/state/common/randomAction';
import { fetchingWeatherApiState } from '../fetchingWeatherApiStateReducer';
import { FETCH_WEATHER_SUCCESS, FETCH_WEATHER_PENDING, FETCH_WEATHER_ERROR } from '../constants';

describe('fetchingWeatherApiState test suite', () => {
    it('should return default state when action not matches', () => {
        const action = randomAction();

        expect(fetchingWeatherApiState(undefined, action)).toMatchSnapshot();
    });

    it('should return default state on success', () => {
        const action = {
            type: FETCH_WEATHER_SUCCESS
        };

        expect(fetchingWeatherApiState(undefined, action)).toMatchSnapshot();
    });

    it('should return state for pending', () => {
        const action = {
            type: FETCH_WEATHER_PENDING
        };

        expect(fetchingWeatherApiState(undefined, action)).toMatchSnapshot();
    });

    it('should return state for error', () => {
        const action = {
            type: FETCH_WEATHER_ERROR
        };

        expect(fetchingWeatherApiState(undefined, action)).toMatchSnapshot();
    });
});