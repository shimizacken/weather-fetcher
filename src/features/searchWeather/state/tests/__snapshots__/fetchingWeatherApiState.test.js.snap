// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchingWeatherApiState test suite should return default state on success 1`] = `
Object {
  "isError": false,
  "isFetching": false,
}
`;

exports[`fetchingWeatherApiState test suite should return default state when action not matches 1`] = `
Object {
  "isError": false,
  "isFetching": false,
}
`;

exports[`fetchingWeatherApiState test suite should return state for error 1`] = `
Object {
  "isError": true,
  "isFetching": false,
}
`;

exports[`fetchingWeatherApiState test suite should return state for pending 1`] = `
Object {
  "isError": false,
  "isFetching": true,
}
`;
